TelegramBot/client/models.py

```diff
@@ -0,0 +1,267 @@
+
+    def add_wishlist(self) -> bool:
+        """
+        Add item to user's wishlist
+        """
+        response = bot_client.send_request("POST", f"user/{self.user_id}/wishlist", data=self.product_id)
+        return response.status_code == 201
+
+    @staticmethod
+    def delete(user_id: int, product_id: int) -> bool:
+        """
+        Delete item from user's wishlist
+        """
+        response = bot_client.send_request("DELETE", f"/user/wishlist", params={"userId": user_id,
+                                                                                "productId": product_id})
+        return response == 200
+
+
+class User(BaseModel):
+    """
+    Model for working with user.
+    self.phone_number field uses PhoneNumber class from phonenumbers library, everything else - Pydantic fields
+    More details on phonenmbers: https://github.com/stefanfoulis/django-phonenumber-field
+    """
+    id: PositiveInt = None
+    first_name: constr(max_length=40)
+    last_name: constr(max_length=40)
+    phone_number: PhoneNumber
+    email: EmailStr
+    birth_date: datetime.date = None
+
+    class Config:
+        """
+        Arbitrary_types_allowed allows us to use non-Pydantic classes without modification for validation
+        """
+        arbitrary_types_allowed = True
+
+    @staticmethod
+    def is_registered(phone_number: PhoneNumber) -> int:
+        """
+        Check if our user is registered by his phone_number. If he is, return his id for later use
+        """
+        response = bot_client.send_request("GET", f"/user", params={"phone-number", phone_number})
+        if response.status_code == 200:
+            return response.json()['id']
+
+    def register(self) -> int:
+        """
+        Send user to save in database. On success, return his dedicated id
+        """
+        response = bot_client.send_request("POST", "/user", data=self.__dict__)
+        if response.status_code == 201:
+            return response.json()["id"]
+
+    @staticmethod
+    def update(user_id: int, data_to_change: dict) -> bool:
+        """
+        Change user_data via dict. Return true on success
+        """
+        response = bot_client.send_request("PATCH", f"/user", params={"userId", user_id}, data=data_to_change)
+        return response.status_code == 200
+
+
+class Category(BaseModel):
+    id: PositiveInt
+    name: constr(max_length=100)
+
+    @classmethod
+    def get(cls) -> list:
+        """Get category list"""
+        response = bot_client.send_request("GET", "/products/categories")
+        if response.status_code == 200:
+            return [cls(**c) for c in response.json()]
+
+
+class Subcategory(BaseModel):
+    id: PositiveInt
+    name: constr(max_length=100)
+
+    @classmethod
+    def get(cls, category_id: int) -> list:
+        """
+        Get subcategory list by category
+        """
+        response = bot_client.send_request("GET", f"/products/categories/{category_id}/subcategories")
```

Comment:
how do we connect category and subcategory if they have no links between each other

Reply:
Yeap, my bad, but this is an issue for client encapsulation branch